project1/server/file_handler/thread_safe_file_handler.cpp

```diff
@@ -0,0 +1,52 @@
+#include "thread_safe_file_handler.h"
+
+#include <utility>
+
+#include "file_lock_guard.h"
+
+namespace server::file_handler {
+
+ThreadSafeFileHandler::ThreadSafeFileHandler(
```

Comment:
I don't really understand this syntax for the constructor. What is this kind of constructor definition called so I can read more documentation about it?

Reply:
What syntax specifically? The [initialization list](https://en.cppreference.com/w/cpp/language/constructor)?